reconcile/external_resources/integration.py

```diff
@@ -141,3 +168,29 @@ def run(
                     logging.error("ExternalResourceKey: %s, Error: %s" % (k, e))
         else:
             er_mgr.handle_resources()
+
+
+def early_exit_desired_state(*args: Any, **kwargs: Any) -> dict[str, Any]:
+    vault_settings = get_app_interface_vault_settings()
+    secret_reader = create_secret_reader(use_vault=vault_settings.vault)
+    er_settings = get_settings()[0]
+
+    with get_aws_api(
+        query_func=gql.get_api().query,
+        account_name=er_settings.state_dynamodb_account.name,
+        region=er_settings.state_dynamodb_region,
+        secret_reader=secret_reader,
+    ) as aws_api:
+        er_mgr = create_er_manager(
+            aws_api,
+            workers_cluster=kwargs["workers_cluster"],
+            workers_namespace=kwargs["workers_namespace"],
+            thread_pool_size=kwargs["thread_pool_size"],
+            dry_run=True,
+            dry_run_job_suffix=kwargs["dry_run_job_suffix"],
+        )
+
+        return {
+            "desired": er_mgr._get_desired_objects_reconciliations(),
+            "deleted": er_mgr._get_deleted_objects_reconciliations(),
```

Comment:
should these 2 methods become public or have a new one for this use case?

Reply:
Yup, it's more correct. PTAL